airflow/providers/fab/provider.yaml

```diff
@@ -25,6 +25,11 @@ description: |
   `Flask App Builder <https://flask-appbuilder.readthedocs.io/>`__
 
 suspended: false
+
+# The provider is not yet ready to be released, we will skip it by default when preparing new release waves
+# For providers until we think it should be released.
+not-ready: true
```

Comment:
I think we might be close to replace all states attributes by a single as enum (as follow up), e.g.:
```json
"state": {
        "type": "string",
        "enum": ["ready", "not-ready", "suspended", "removed"]
      },
```
I think all states are mutually exclusive and lifecycle for abstract package would be

```mermaid
graph TD;
    not-ready-->ready;
    ready-->suspended;
    ready-->removed;
    suspended-->removed;
```

Reply:
Yes. We could do it this way as well. I will leave that exercise to you :)

Reply:
It's a good idea

Reply:
BTW. Here is the PR about state updating by Release Manager https://github.com/apache/airflow/pull/36392

Reply:
BTW. I like the diagram and added it to the docs now. But I  **fixed** it. There is one more transition possible - ready -> not-ready. The way we have it defined now, we can mark a package as "not-ready" even if it has been previously released and it will be all-fine with tests and ci - but it will be skipped by default when releasing new wave.

Reply:
And of course `suspended->ready` is also possible.

Reply:
Looks a bit different now :)

<img width="398" alt="Screenshot 2023-12-23 at 17 49 40" src="https://github.com/apache/airflow/assets/595491/0435d2fd-94ad-4772-a9b2-749e3fb85473">

Reply:
That is a bit TOO much though 👎 

<img width="1105" alt="Screenshot 2023-12-23 at 17 59 14" src="https://github.com/apache/airflow/assets/595491/40ce594f-51ba-4716-88c4-2d849881bbb9">

Reply:
How about this ? 

![Uploading Screenshot 2023-12-23 at 18.03.06.png…]()

Reply:
Ok. This one seems to be nicest: 

<img width="690" alt="Screenshot 2023-12-23 at 18 05 24" src="https://github.com/apache/airflow/assets/595491/ced74b12-ba58-4b33-87cf-8e61cb775b6f">

Reply:
Or this for symmetry


![Uploading Screenshot 2023-12-23 at 18.07.54.png…]()